MuvrKit/MKClassifier.swift

```diff
@@ -93,40 +106,116 @@ public struct MKClassifier {
                 
                 let labelName = self.model.exerciseIds[labelIndex]
                 let probability = windowPrediction[labelIndex]
+                
                 let windowDuration = window == numWindows - 1 ? (Double(windowSize) / Double(block.samplesPerSecond)) : duration
                 return MKClassifiedExerciseBlock(confidence: Double(probability), exerciseId: labelName, duration: windowDuration, offset: start)
             }
             return MKClassifiedExerciseWindow(window: window, classifiedExerciseBlocks: classifiedExerciseBlocks)
         }
-        
+        return cews
+    }
+    
+    ///
+    /// Compute the probability for every window step by averaging the probabilities over all windows containing the step
+    ///
+    func classifySteps(windows: [MKClassifiedExerciseWindow], samplesPerSecond: UInt) -> [MKClassifiedExerciseBlock] {
+        let duration = Double(windowStepSize) / Double(samplesPerSecond)
```

Comment:
Won't the duration be wrong for the last window? Since the last window will be 8 seconds long?